src/main/scala/pipedsl/typechecker/LockConstraintChecker.scala

```diff
@@ -211,7 +211,7 @@ class LockConstraintChecker(lockMap: Map[Id, Set[LockArg]], lockGranularityMap:
       val env2 = checkExpr(tval, env1, predicates)
       checkExpr(fval, env2, predicates)
     case EApp(_, args) => args.foldLeft(env)((e, a) => checkExpr(a, e, predicates))
-    case ECall(_, _, args) => args.foldLeft(env)((e, a) => checkExpr(a, e, predicates))
+    case ECall(_, _, args, _) => args.foldLeft(env)((e, a) => checkExpr(a, e, predicates))
```

Comment:
We also need to check that the lock for this pipeline is acquired if the mem access isn't atomic here too